sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java

```diff
@@ -162,32 +181,42 @@
      *
      * <p>Add a setting with the key "prodDBConnection" and value "db_connection".</p>
      *
-     * <pre>
-     * client.setSetting("prodDBConnection", "db_connection")
-     *     .subscribe(response -&gt; {
-     *         ConfigurationSetting result = response.value();
-     *         System.out.printf("Key: %s, Value: %s", result.key(), result.value());
-     *     });</pre>
+     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.setSettingWithResponse#ConfigurationSetting}
+     *
+     * @param key The key of the configuration setting to create or update.
+     * @param value The value of this configuration setting.
+     * @return The {@link ConfigurationSetting} that was created or updated, or {@code null}, if the key is an invalid
+     * value (which will also throw HttpResponseException described below).
+     * @throws IllegalArgumentException If {@code key} is {@code null}.
+     * @throws ResourceModifiedException If the setting exists and is locked.
+     * @throws HttpResponseException If {@code key} is an empty string.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<ConfigurationSetting> setSetting(String key, String value) {
+        return monoContext(
+            context -> setSetting(new ConfigurationSetting().key(key).value(value), context))
+            .flatMap(response -> Mono.justOrEmpty(response.value()));
+    }
+
+    /**
+     * Creates or updates a configuration value in the service with the given key.
+     *
+     * <p><strong>Code Samples</strong></p>
      *
-     * <p>Update the value of the setting to "updated_db_connection".</p>
+     * <p>Add a setting with the key "prodDBConnection" and value "db_connection".</p>
      *
-     * <pre>
-     * client.setSetting("prodDBConnection", "updated_db_connection")
-     *     .subscribe(response -&gt; {
-     *         ConfigurationSetting result = response.value();
-     *         System.out.printf("Key: %s, Value: %s", result.key(), result.value());
-     *     });</pre>
+     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.setSettingWithResponse#string-string}
      *
      * @param key The key of the configuration setting to create or update.
      * @param value The value of this configuration setting.
-     * @return The {@link ConfigurationSetting} that was created or updated, or {@code null}, if the key is an invalid
+     * @return The {@link ConfigurationSetting} that was created or updated with the REST response, or {@code null}, if the key is an invalid
```

Comment:
For the WithResponse methods it would read more correctly as:

`@return A REST response containing the {@link ConfigurationSetting} that was created or updated, or {@code null if the key is an invalid....`